src/python/pants/engine/legacy/graph.py

```diff
@@ -755,15 +724,14 @@ def _eager_fileset_with_spec(
         logger.warning(msg)
       else:
         raise ResolveError(msg)
-    for address in owners.addresses:
-      # We preserve what literal files any globs resolved to. This allows downstream goals to be
-      # more precise in which files they operate on.
-      origin = (
-        spec
-        if isinstance(spec, FilesystemLiteralSpec) else
-        FilesystemResolvedGlobSpec(glob=spec.glob, resolved_files=snapshot.files)
-      )
-      result.append(AddressWithOrigin(address=address, origin=origin))
+    # We preserve what literal files any globs resolved to. This allows downstream goals to be
+    # more precise in which files they operate on.
+    origin = (
+      spec
+      if isinstance(spec, FilesystemLiteralSpec) else
+      FilesystemResolvedGlobSpec(glob=spec.glob, resolved_files=snapshot.files)
+    )
+    result.extend(AddressWithOrigin(address, origin) for address in owners.addresses)
```

Comment:
Optimization of prior PR. Realized it wasn't as efficient or readable as it could have been.